From 011b8ed44e3648cf5c70cd216aa1696b8fbdc95a Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Mon, 13 Sep 2010 20:36:07 +0800
Subject: [PATCH 126/249] ENGR00131414 ipu lib: fix segmentation fault while
 call kill control

During kill control, fb should not be filled.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 17 +++++++++++++----
 1 file changed, 13 insertions(+), 4 deletions(-)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index d2fe0b3..58b8f63 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -2631,11 +2631,13 @@ static void _mxc_ipu_lib_task_uninit(ipu_lib_priv_handle_t * ipu_priv_handle, pi
 			struct fb_fix_screeninfo fb_fix;
 			struct fb_var_screeninfo fb_var;
 
-			ioctl(ipu_priv_handle->output.fd_fb, FBIOGET_FSCREENINFO, &fb_fix);
-			ioctl(ipu_priv_handle->output.fd_fb, FBIOGET_VSCREENINFO, &fb_var);
+			if (!kill) {
+				ioctl(ipu_priv_handle->output.fd_fb, FBIOGET_FSCREENINFO, &fb_fix);
+				ioctl(ipu_priv_handle->output.fd_fb, FBIOGET_VSCREENINFO, &fb_var);
 
-			__fill_fb_black(ipu_priv_handle->output.ofmt, &fb_var, &fb_fix,
-					ipu_priv_handle->output.fb_mem);
+				__fill_fb_black(ipu_priv_handle->output.ofmt, &fb_var, &fb_fix,
+						ipu_priv_handle->output.fb_mem);
+			}
 
 			if ( ioctl(ipu_priv_handle->output.fd_fb, MXCFB_SET_OVERLAY_POS,
 						&pos) < 0)
@@ -3291,6 +3293,13 @@ int mxc_ipu_lib_task_control(int ctl_cmd, void * arg, ipu_lib_handle_t * ipu_han
 {
 	int ret = 0;
 	ipu_lib_priv_handle_t * ipu_priv_handle;
+	char * dbg_env;
+
+	dbg_env = getenv("IPU_DBG");
+	if (dbg_env) {
+		debug_level = atoi(dbg_env);
+		dbg(DBG_INFO, "ipu debug level %d\n", debug_level);
+	}
 
 	if (ipu_handle)
 		ipu_priv_handle = (ipu_lib_priv_handle_t *)ipu_handle->priv;
-- 
1.8.0

